### DISCUSSION TITLE
Error when a shortcode and a paired shortcode are registered with the same name

### DISCUSSION THREAD

**Discussion by LeaVerou** [11ty/eleventy#4098] [Enhancement Queue]
E.g.

```js
config.addPairedShortcode("scoped", scope);
config.addShortcode("scoped", scope);
```

produces `Error: unknown block tag: endscoped` if the paired version is used. The error is the same regardless of the order the shortcodes are registered in. 

Note that having both types of shortcodes with the same name is what Nunjucks itself does for `{% set %}`.

To avoid having to parse a ton of content to figure out which one is being used, perhaps it could be smart about which arguments are being used (not a fully fleshed idea, just throwing it out there).

**Comment by zachleat** [comment]
You’re right that the shortcodes API does not currently support combining paired and unpaired shortcodes together. Going to move this one to the enhancement queue.

In the mean time—behind the scenes, both use Nunjucks’ `addExtension` API, which is currently exposed as a lower level piece using Custom Tags, so you _can_ likely wire something up here yourself: https://www.11ty.dev/docs/custom-tags/

> By default this repo closes enhancement request issues and places them in a queue for folks to upvote. We do this to avoid a large build-up of open issues. We got this idea from the [lodash project](https://twitter.com/samselikoff/status/991395669016436736).

> Don’t forget to upvote the first comment with 👍 to register your vote! You can view [the current enhancement backlog](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+).

**Comment by LeaVerou** [comment]
Yeah I suspected as much. Bummer.

Unrelated, have you considered using Discussions for enhancements? Then no need to close anything, which makes issues hard to find, since search only targets open issues by default.
